Kickstarter-iOS/Features/Messages/Controller/MessagesViewController.swift

```diff
@@ -161,6 +194,48 @@ internal final class MessagesViewController: UITableViewController, MessageBanne
     self.present(vc, animated: true)
   }
 
+  private func updateTableViewBottomContentInset() {
+    if let userBlockedBannerHostingController = userBlockedBannerHostingController {
+      self.tableView.contentInset
+        .bottom = userBlockedBannerHostingController.view.bounds
+        .height
+    }
+  }
+
+  private func removeUserBlockedBanner() {
+    self.userBlockedBannerHostingController?.removeFromParent()
+
+    let userBlockedBannerView = self.tabBarController?.view.subviews.first(where: { view in
+      view.viewWithTag(-99) != nil
+      })
+
+    userBlockedBannerView?.removeFromSuperview()
+  }
+
+  private func showUserBlockedBanner() {
+    let userBlockedBannerHostingController = UIHostingController(rootView: UserBlockedBannerView())
+
+    guard let tabController = self.tabBarController as? RootTabBarViewController,
+      let userBlockedBannerView = userBlockedBannerHostingController.view else { return }
+
+    userBlockedBannerView.tag = -99
+    tabController.addChild(userBlockedBannerHostingController)
+    tabController.view.addSubview(userBlockedBannerView)
```

Comment:
adding the banner to the tabController so rather than the viewcontroller directly. This helped to avoid some more specific styling that we would need to do in order to have this banner sit above the tab bar and not block the ViewController's content.

With this approach we can simply call `updateTableViewBottomContentInset` on viewDidAppear.